fontc/src/main.rs

```diff
@@ -414,30 +415,65 @@ fn add_glyph_merge_be_job(
         let write_access: AccessFn<_> = Arc::new(|id| {
             matches!(
                 id,
-                AnyWorkId::Be(BeWorkIdentifier::GlyphMerge(..))
+                AnyWorkId::Be(BeWorkIdentifier::Glyf)
+                    | AnyWorkId::Be(BeWorkIdentifier::Loca)
                     | AnyWorkId::Be(BeWorkIdentifier::Glyph(..))
             )
         });
-        let id: AnyWorkId = BeWorkIdentifier::GlyphMerge(GlyphMerge::Glyf).into();
+        let id: AnyWorkId = BeWorkIdentifier::Glyf.into();
         workload.insert(
             id,
             Job {
-                work: create_glyph_merge_work().into(),
+                work: create_glyf_loca_work().into(),
                 dependencies,
                 // We need to read all glyphs, even unchanged ones, plus static metadata
                 read_access: ReadAccess::Custom(Arc::new(|id| {
                     matches!(
                         id,
-                        AnyWorkId::Be(BeWorkIdentifier::GlyphMerge(..))
+                        AnyWorkId::Be(BeWorkIdentifier::Glyf)
+                            | AnyWorkId::Be(BeWorkIdentifier::Loca)
                             | AnyWorkId::Be(BeWorkIdentifier::Glyph(..))
                     )
                 })),
                 write_access,
             },
         );
     } else {
-        workload.mark_success(BeWorkIdentifier::GlyphMerge(GlyphMerge::Glyf));
-        workload.mark_success(BeWorkIdentifier::GlyphMerge(GlyphMerge::Loca));
+        workload.mark_success(BeWorkIdentifier::Glyf);
+        workload.mark_success(BeWorkIdentifier::Loca);
+    }
+    Ok(())
+}
+
+fn add_cmap_be_job(
+    change_detector: &mut ChangeDetector,
+    workload: &mut Workload,
+) -> Result<(), Error> {
+    let glyphs_changed = change_detector.glyphs_changed();
+
+    // If no glyph has changed there isn't a lot of merging to do
+    if !glyphs_changed.is_empty() {
+        let mut dependencies: HashSet<_> = glyphs_changed
+            .iter()
+            .map(|gn| FeWorkIdentifier::Glyph(gn.clone()).into())
+            .collect();
+        dependencies.insert(FeWorkIdentifier::FinalizeStaticMetadata.into());
+
+        let id: AnyWorkId = BeWorkIdentifier::Cmap.into();
+        workload.insert(
+            id.clone(),
+            Job {
+                work: create_cmap_work().into(),
+                dependencies,
+                // We need to read all glyph IR, even unchanged ones, plus static metadata
```

Comment:
> We need to read all glyph IR

if you had the global {character: glyph} mappings, you wouldn't need to read all glyph IRs, just the static metadata

Reply:
Agreed, I have forked that discussion to https://github.com/googlefonts/fontmake-rs/issues/145